chingu-voyages/v37-bears-team-14
# Adds webhooks for getting repository events from GitHub

# Changes

These changes help to capture project events for tracking user contributions.

- Adds Hook and ProjectEvent model for creating GitHub webhooks
- Adds project settings page for creating new webhooks

# Testing

- Run app
- Go to a project you own
- Create a webhook
- Add the webhook to a github repository

See https://docs.github.com/en/developers/webhooks-and-events/webhooks/creating-webhooks for proxying with ngrok for local testing. Otherwise, `backend/scripts/testEventGithub.ts` mocks the GitHub call.
